bcel-builder/src/main/java/org/aspectj/apache/bcel/util/ClassLoaderRepository.java

```diff
@@ -326,6 +349,7 @@ public JavaClass loadClass(String className) throws ClassNotFoundException {
 			classesLoadedCount++;
 			return clazz;
 		} catch (IOException e) {
+			unavailableClasses.add(className);
```

Comment:
For 100% backward compatibility, this statement should be shielded like this, just like you did it above:
```java
if (useUnavailableClassesCache) {
  unavailableClasses.add(className);
}
```